### PR TITLE
Stop prefixing legacy URLs with /__browser.legacy

### PR DESCRIPTION
**Pull request by benjamn**
Inspired by discussion with @CaptainN in #9776, I think we can (and should) stop relying on URL prefixes to determine whether to load static files from the `web.browser` (modern) or `web.browser.legacy` (legacy) architectures. If this works, it will be a major ergonomic simplification for the modern/legacy system coming in Meteor 1.7.

This PR makes `webapp` serve different static files based on `isModern(request.browser)`, which is something we already have to do without the help of URL prefixes [in the `dynamic-import` package](https://github.com/meteor/meteor/blob/8e6f673fd066a69b2b4779526d200a7be0f1fc73/packages/dynamic-import/server.js#L80-L82).

Note that `web.cordova` URLs will still be prefixed with `/__cordova/`, as they always were, even before #9439 split `web.browser` and `web.browser.legacy`. Additionally, URL prefixes like `/__browser/` and `/__browser.legacy/` can still be used explicitly to request resources from non-default architectures.

It feels a little weird for the server to respond with different resources for the same URL, depending on the user agent string, but any given client will only see one version of the resource for a given URL, so clients can still safely cache the resources they receive. Furthermore, JS and CSS resources are always requested using a hash of the resource's contents (in both development and production), so the modern and legacy URLs for those kinds of resources will be different anyway.

### CODE REVIEW

File: packages/webapp/webapp_server.js [hunk 1]
```diff
@@ -467,6 +471,63 @@ WebAppInternals.staticFilesMiddleware = function (staticFiles, req, res, next) {
   }
 };
 
+function getStaticFileInfo(originalPath, browser) {
+  const { arch, path } = getArchAndPath(originalPath, browser);
+
+  if (! hasOwn.call(WebApp.clientPrograms, arch)) {
+    return null;
```
Comment (benjamn) [inline packages/webapp/webapp_server.js]: This check may be redundant with the one below.

**Comment by CaptainN** [comment]
This looks reasonable. We should make sure to document the difference in deliverables, so that front end caches can make sure to set 'vary-by' to for useragent request headers. An alternative idea - would it make sense to add a legacy argument to the query string?

How many assets actually require a legacy URL? I'm under the impression that only built assets would need the legacy modifier - the main JS bundles (maybe worker bundles if we add those at some point) and maybe the CSS file, if there's benefit to stripping out auto-prefixer overhead (maybe not). What else might need legacy URLs?

File: packages/non-core/bundle-visualizer/server.js [hunk 2]
```diff
@@ -33,21 +33,38 @@ function getStatBundles() {
...
+  }
+
+  const {
+    staticFiles,
+    staticFilesByArch,
+  } = WebAppInternals;
+
+  const files = [];
+
+  if (staticFilesByArch) {
```
Comment (hwillson) [inline packages/non-core/bundle-visualizer/server.js]: Total drive-by comment, but you might be able to `DRY` this up a bit with something like:

```js
if (staticFilesByArch || staticFiles) {
  const sourceFiles = staticFilesByArch || { '': staticFiles };
  Object.keys(sourceFiles).forEach(arch => {
    const staticFiles = sourceFiles[arch];
    Object.keys(staticFiles).forEach(path => files.push(staticFiles[path]));
  });
}
```
